sentinelsat/sentinel.py

```diff
@@ -145,13 +144,13 @@ def query(self, area, initial_date=None, end_date=datetime.now(), **keywords):
         and any other search keywords accepted by the SciHub API.
         """
         query = self.format_query(area, initial_date, end_date, **keywords)
-        self.load_query(query)
-        return self.products
+        return self.load_query(query)
 
-    def load_query(self, query, start_row=0):
+    def load_query(self, query, start_row=0, products=[]):
         """Do a full-text query on the SciHub API using the format specified in
            https://scihub.copernicus.eu/twiki/do/view/SciHubUserGuide/3FullTextSearch
         """
+        output = products
```

Comment:
Ok, fixed in fc7fe1e
Now most of the tests pass, only two are still failing, `test_large_query` because  `assert len(products) > api.page_size` seems to be `100>100`; instead I'm not able to reproduce the error in `test_download_all`, if I follow the same code in a Python console it works.